### PR TITLE
Add Nested Nil Checks To Wrapper

### PR DESCRIPTION
**Pull request by nisdas**
**What type of PR is this?**

Cleanup PR

**What does this PR do? Why is it needed?**

In #8918, @rkapka brought up the fact that we do not perform checks on the inner structures of our block objects. Given that the interface PR was big and touching a lot of places, it was decided it would be better to add in the checks in a follow up PR. The PR *should* not break anything, as we should always be handling a non-nil block,body,etc, but given the many places a block object is used there maybe edge cases where this isnt accounted for. This PR adds in nil checks to the wrapper methods. 

**Which issues(s) does this PR fix?**

N.A

**Other notes for review**

### CODE REVIEW

File: shared/interfaces/phase0_block_wrapper.go [hunk 1]
```diff
@@ -33,7 +33,7 @@ func (w Phase0SignedBeaconBlock) Block() BeaconBlock {
 // IsNil checks if the underlying beacon block is
 // nil.
 func (w Phase0SignedBeaconBlock) IsNil() bool {
-	return w.b == nil
+	return w.b == nil || w.Block().IsNil()
```
Comment (rkapka) [inline shared/interfaces/phase0_block_wrapper.go:36]:
```suggestion
	return w.b == nil || w.Block() == nil || w.Block().IsNil()
```

File: shared/interfaces/phase0_block_wrapper.go [hunk 2]
```diff
@@ -97,7 +97,7 @@ func (w Phase0BeaconBlock) Body() BeaconBlockBody {
 
 // IsNil checks if the beacon block is nil.
 func (w Phase0BeaconBlock) IsNil() bool {
-	return w.b == nil
+	return w.b == nil || w.Body().IsNil()
```
Comment (rkapka) [inline shared/interfaces/phase0_block_wrapper.go:100]:
```suggestion
	return w.b == nil || w.Body() == nil || w.Body().IsNil()
```

**Review by rkapka — CHANGES_REQUESTED** [review]

File: shared/interfaces/phase0_block_wrapper.go (hunk 1 above)
Comment (nisdas) [reply shared/interfaces/phase0_block_wrapper.go:36]: `w.Block() == nil` would never be true. If you look at our `Block` method, we always use a wrapper around the block proto object. So its impossible for the interface to be nil.

File: shared/interfaces/phase0_block_wrapper.go (hunk 2 above)
Comment (nisdas) [reply shared/interfaces/phase0_block_wrapper.go:100]: Same here

**Review by rkapka — APPROVED** [review]

**Pull request merged**